### PR TITLE
Flex all layouts - first pass

### PR DESCRIPTION
**Pull request by mor10**
First pass replacing all float-based layout positioning with flex(box) for easier management and cleaner RTL style rules. Addresses #16 

Also includes some minimal CSS and markup cleanup.

Right now all flexed items revert to `display:block` (browser default) when flex is not supported causing a vertical layout. To ensure backward compat to older browsers and IE8/9 we should include Flexibility (https://github.com/jonathantneal/flexibility) with a conditional to test for those older browsers so we don't load an unnecessary JS library for current browsers.

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -3338,4 +3348,4 @@ article.panel-placeholder {
 	.twentyseventeen-front-page .entry-content blockquote.alignright {
 		margin-right: -20%;
 	}
-}
\ No newline at end of file
+}
```
Comment (ntwb) [inline style.css]: The new blank line at the end of the file shouldn't be removed

File: style.css [hunk 2]
```diff
@@ -3228,6 +3232,7 @@ article.panel-placeholder {
 	.nav-links .nav-next .nav-title:after {
 		color: #222;
 		position: absolute;
+		top: -.2em;
```
Comment (ntwb) [inline style.css]: Expected a leading zero, `top: -0.2em;`, not `top: -.2em;`

**Comment by ntwb** [comment]
@mor10 Could you please add `assets/js/flexibility.js` to both the following files so that this file is excluded from the JSHint and JSCS tests:
• https://github.com/WordPress/twentyseventeen/blob/master/.jscsrc
• https://github.com/WordPress/twentyseventeen/blob/master/.jshintignore

**Comment by ntwb** [comment]
There is also ~15 PHP coding standards errors, you can see the list of these in this Travis CI job:
https://travis-ci.org/WordPress/twentyseventeen/jobs/166493519#L688

**Comment by mor10** [comment]
I have no idea why they are showing up. The only change to the current PR
is the enqueueing of flexibility in functions.php and custom prefix
references in style.css. The errors in Travis appear to be other unrelated
things.

**Comment by ntwb** [comment]
@mor10 Right you are for the PHP, apologies.

The `assets/js/flexibility.js` file still needs to be added to the `.jshintignore` and `.jscsrc` files though so the JSCS and JSHint checks ignore it.

**Comment by mor10** [comment]
Will do, first thing in the morning.

**Comment by mor10** [comment]
@josephfusco What refresh do you want? The conflict seems to be the files @ntwb asked me to add in. Do you want me to take them out again?

**Comment by josephfusco** [comment]
@mor10 Yes I just meant those files which I see you added but it's still conflicting.

This might require a rebase to resolve.

**Comment by mor10** [comment]
This is what happens when PRs take too long to merge. I will rebase and make the necessary updates to get rid of the conflicts. Updated PR will not come until late Thursday at the earliest.

**Comment by mor10** [comment]
PR is refreshed, rebased, and ready to go.

**Review by josephfusco — APPROVED** [review]

**Pull request closed without merging**